### PR TITLE
Fixed screeninfo.get_monitors returning empty on osx

### PR DESCRIPTION
**Pull request by betafcc**
## Overview: What does this pull request change?
Fixes screening.get_monitors returning empty on OSX

Issue: https://github.com/ManimCommunity/manim/issues/2356

## Motivation and Explanation: Why and how do your changes improve the library?

At the moment, the quick start example does not work for `--renderer=opengl`, this fixes it

## Further Information and Comments

The problem is also present in maningl: https://github.com/3b1b/manim/issues/1394

Seems it * should * be handled by https://github.com/rr-/screeninfo , but some probably related issues are open since mid-2020.

The get_monitors function seem to try all possible platforms when one is not supplied explicitly:

https://github.com/rr-/screeninfo/blob/master/screeninfo/screeninfo.py#L20-L35

But you can see in these lines:
```py
    for enumerator in Enumerator:
        try:
            return _get_monitors(enumerator)
        except Exception:
            pass
```
it tries the next one only if previous raised exception, I suspect some enumerator that is tested before osx is returning successfully when it shouldn't.

I will investigate and try a PR on screeninfo, maybe I'll make it check and fail also for empty lists, but even if it makes it through, last release there was almost a year ago, will take some time to fix us

## Reviewer Checklist
- [ ] The PR title is descriptive enough for the changelog, and the PR is labeled correctly
- [ ] If applicable: newly added non-private functions and classes have a docstring including a short summary and a PARAMETERS section
- [ ] If applicable: newly added functions and classes are tested

### CODE REVIEW

File: manim/renderer/opengl_renderer_window.py [hunk 1]
```diff
@@ -14,7 +16,10 @@ class Window(PygletWindow):
     cursor = True
 
     def __init__(self, renderer, size=config.window_size, **kwargs):
-        monitors = get_monitors()
+        if platform.system() == "Darwin":
```
Comment (ryanmccauley211) [inline manim/renderer/opengl_renderer_window.py:19]: This seems fine to me if we patch this, but I know @naveen521kk was saying it should be fixed upstream. This PR seems to fix it and is approved so might be better waiting for screeninfo to merge and not add extra logic to manim ? https://github.com/rr-/screeninfo/pull/61

Comment (betafcc) [reply manim/renderer/opengl_renderer_window.py:19]: yes! Indeed @rr- was quick to answer and fix it, we will be releasing soon, then I'll revert this commit and just update the dependency

Comment (behackl) [reply manim/renderer/opengl_renderer_window.py:19]: Great! I think waiting for the fix from upstream is the correct approach here; I'll convert this PR into a draft in the meantime. Thank you for your contribution!

Comment (rr-) [reply manim/renderer/opengl_renderer_window.py:19]: I just released docstring-parser 0.8 which should fix this issue. Thank you @betafcc for great contributions :)

**Comment by betafcc** [comment]
@ryanmccauley211 sorry I'm late, holiday season is quite busy.
The update makes opengl work fine in macos. I did got a failed test though, but also tried in main and it's there as well, so don't know what's of it:

<details>
<summary>pytest output</summary>

```sh
manim/utils/testing/frames_comparison.py:128:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
manim/utils/testing/frames_comparison.py:218: in real_test
    frames_tester.check_frame(-1, scene_tested.renderer.get_frame())
manim/utils/testing/_frames_testers.py:52: in check_frame
    raise e
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <manim.utils.testing._frames_testers._FramesTester object at 0x12cd4f6d0>, frame_number = -1
frame = array([[[  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        ...,
        [  0,  ...,
        ...,
        [  0,   0,   0, 255],
        [  0,   0,   0, 255],
        [  0,   0,   0, 255]]], dtype=uint8)

    def check_frame(self, frame_number: int, frame: np.ndarray):
        assert frame_number < self._number_frames, (
            f"The tested scene is at frame number {frame_number} "
            f"when there are {self._number_frames} control frames."
        )
        try:
>           np.testing.assert_allclose(
                frame,
                self._frames[frame_number],
                atol=1.01,
                err_msg=f"Frame no {frame_number}. You can use --show_diff to visually show the difference.",
                verbose=False,
            )
E           AssertionError:
E           Not equal to tolerance rtol=1e-07, atol=1.01
E           Frame no -1. You can use --show_diff to visually show the difference.
E           Mismatched elements: 6 / 1639680 (0.000366%)
E           Max absolute difference: 255
E           Max relative difference: 1.94656489

manim/utils/testing/_frames_testers.py:41: AssertionError
------------------------------------------------ Captured stdout call ------------------------------------------------
(1, 480, 854, 4)
================================================== warnings summary ==================================================
tests/test_value_tracker.py: 16 warnings
tests/test_graphical_units/test_axes.py: 5 warnings
tests/test_graphical_units/test_mobjects.py: 5 warnings
tests/test_graphical_units/test_threed.py: 75 warnings
tests/test_graphical_units/test_updaters.py: 1 warning
  /Users/brendon.faccion/Desktop/meera/manim/manim/mobject/value_tracker.py:72: DeprecationWarning: This method is not guaranteed to stay around. Please prefer setting the attribute normally or with Mobject.set().
    self.set_points(np.zeros((1, 3)))

tests/test_value_tracker.py: 1 warning
tests/test_graphical_units/test_threed.py: 24 warnings
  /Users/brendon.faccion/Desktop/meera/manim/manim/mobject/value_tracker.py:133: DeprecationWarning: This method is not guaranteed to stay around. Please prefer setting the attribute normally or with Mobject.set().
    self.set_points(path_func(mobject1.points, mobject2.points, alpha))

tests/opengl/test_axes_shift_opengl.py: 4 warnings
tests/opengl/test_coordinate_system_opengl.py: 68 warnings
tests/opengl/test_opengl_vectorized_mobject.py: 1 warning
tests/opengl/test_ticks_opengl.py: 12 warnings
  /Users/brendon.faccion/Desktop/meera/manim/manim/mobject/types/opengl_vectorized_mobject.py:923: DeprecationWarning: Calling np.sum(generator) is deprecated, and in the future will give a different result. Use np.sum(np.fromiter(generator)) or the python sum builtin instead.
    target_length = alpha * np.sum(length for _, length in curves_and_lengths)

tests/test_graphical_units/test_axes.py::test_get_area
tests/test_graphical_units/test_axes.py::test_get_area
tests/test_graphical_units/test_axes.py::test_get_area
tests/test_graphical_units/test_axes.py::test_get_area
tests/test_graphical_units/test_axes.py::test_get_area
tests/test_graphical_units/test_axes.py::test_get_area
tests/test_graphical_units/test_axes.py::test_get_area
  /Users/brendon.faccion/Desktop/meera/manim/manim/mobject/number_line.py:424: DeprecationWarning: This method is not guaranteed to stay around. Please prefer getting the attribute normally.
    num_mob.shift(num_mob[0].get_width() * LEFT / 2)

tests/test_graphical_units/test_specialized.py::test_Broadcast
tests/test_graphical_units/test_specialized.py::test_Broadcast
tests/test_graphical_units/test_specialized.py::test_Broadcast
tests/test_graphical_units/test_specialized.py::test_Broadcast
tests/test_graphical_units/test_specialized.py::test_Broadcast
  /Users/brendon.faccion/Desktop/meera/manim/manim/animation/specialized.py:85: DeprecationWarning: This method is not guaranteed to stay around. Please prefer setting the attribute normally or with Mobject.set().
    mob.set_width(self.initial_width)

-- Docs: https://docs.pytest.org/en/stable/warnings.html
============================================== short test summary info ===============================================
FAILED tests/test_graphical_units/test_threed.py::test_Axes - AssertionError:
============== 1 failed, 559 passed, 2 skipped, 7 xfailed, 2 xpassed, 224 warnings in 183.41s (0:03:03) ==============
```

</details>

**Comment by betafcc** [comment]
@ryanmccauley211 I don't get it, it failed before reaching the tests, did my PR changed something on the bootstrap config? O.o, I can't see how

ps: please note I rebased with this repository main branch before my last commit, is it something related to that?

**Comment by behackl** [comment]
> it failed before reaching the tests, did my PR changed something on the bootstrap config? O.o, I can't see how

No, the failures seem to be unrelated. I'll re-run the pipeline.

**Review by Darylgolden — APPROVED** [review]
Sorry for the long wait! This LGTM. Can you fix the merge conflicts?

**Review by behackl — APPROVED** [review]

**Pull request merged**